demo/lightning/checkpoint/multinode/train.py

```diff
@@ -0,0 +1,236 @@
+import io
+import os
+import socket
+import time
+import torch
+
+from contextlib import contextmanager
+from typing import Generator
+from lightning import Trainer
+from lightning.pytorch.callbacks import ModelCheckpoint
+from lightning.pytorch.demos import (LightningTransformer, Transformer,
+                                     WikiText2)
+from torch.utils.data import DataLoader
+from pathlib import Path
+from google.cloud import storage
+
+from dataflux_pytorch.lightning.gcs_filesystem import GCSDistributedWriter, GCSDistributedReader
+from dataflux_pytorch.lightning.path_utils import parse_gcs_path
+from dataflux_core import user_agent
+from dataflux_pytorch.lightning import DatafluxLightningCheckpoint
+from lightning.pytorch.trainer.states import TrainerFn
+from lightning.pytorch.strategies import FSDPStrategy
+from lightning.pytorch.strategies.fsdp import _METADATA_FILENAME
+from torch.distributed.checkpoint import save, load
+from torch.nn import Module
+
+
+class DatafluxFSDPStrategy(FSDPStrategy):
```

Comment:
Is the long-term plan for this code to leave it as a demo or actually make it a part of the Dataflux library code eventually?

Reply:
That is a good question, although answer to this really depends on the customer use case. My though process is, if the customers are mostly going to use either FSDP or DDP strategy then we can provide custom classes which extends FSDP/DDP. 

This is something that needs some discussion, but for now I plan to keep this here in the demo, so that it can be added to continuous testing.